src/object.d

```diff
@@ -1930,14 +1934,11 @@ class TypeInfo_Struct : TypeInfo
         assert(p);
         if (xtoHash)
         {
-            return (*xtoHash)(p);
+            return hashOf((*xtoHash)(p));
```

Comment:
If https://github.com/dlang/druntime/pull/2246 is accepted you will be able to revert this to `return (*xtoHash)(p);`